pkgs/test_core/lib/src/runner/configuration/args.dart

```diff
@@ -177,12 +177,22 @@ Configuration parse(List<String> args) => _Parser(args).parse();
 
 void _parseTestSelection(
     String option, Map<String, Set<TestSelection>> selections) {
+  // If given a path that starts with what looks like a drive letter, convert it
+  // into a file scheme URI. We can't parse using `Uri.file` because we do
+  // support query parameters which aren't valid file uris.
+  if (option.indexOf(':') == 1) {
+    option = 'file:///$option';
```

Comment:
Should the slashes be swapped too? This could result in `file:///D:\foo\bar.dart` which seems a little odd (even if it works). 

Probably should only do this on Windows too, since I think colons may be valid on other platforms in filenames? (I can `touch a:b.dart` on mac Mac and it creates that file fine 😄)

Reply:
> Should the slashes be swapped too? This could result in file:///D:\foo\bar.dart which seems a little odd (even if it works).

It works just fine (it is a bit weird, but it works, I am trying to avoid as much manual munging of things as possible here, and just leave it up to the core libs). 

> Probably should only do this on Windows too, since I think colons may be valid on other platforms in filenames? (I can touch a:b.dart on mac Mac and it creates that file fine 😄)

Wow, lol. I guess so.